File: src/main/java/no/nav/altinnkanal/services/TopicService.java
package no.nav.altinnkanal.services;

import no.nav.altinnkanal.entities.TopicMapping;

public interface TopicService {
    TopicMapping getTopicMapping(String serviceCode, String serviceEditionCode) throws Exception;

    void createTopicMapping(String serviceCode, String serviceEditionCode, String topic, long logEntry, Boolean enabled) throws Exception;

    void updateTopicMapping(String serviceCode, String serviceEditionCode, String topic, long logEntry, Boolean enabled) throws Exception;
}

File: altinnkanal/src/main/kotlin/no/nav/altinnkanal/services/TopicService.kt
package no.nav.altinnkanal.services

import no.nav.altinnkanal.config.TopicRouting
import no.nav.altinnkanal.config.loadAivenRouting
import no.nav.altinnkanal.config.loadTopicRouting

open class TopicService constructor(config: TopicRouting = loadTopicRouting()) {
    private val routes: List<TopicRouting.TopicRoute> = config.routes

    open fun getTopics(serviceCode: String, serviceEditionCode: String): List<String>? =
        routes.find { it.serviceCode == serviceCode && it.serviceEditionCode == serviceEditionCode }?.topics
}

open class AivenTopiccService constructor(config: TopicRouting = loadAivenRouting()) {
    private val routes: List<TopicRouting.TopicRoute> = config.routes

    open fun getTopics(serviceCode: String, serviceEditionCode: String): List<String>? =
        routes.find { it.serviceCode == serviceCode && it.serviceEditionCode == serviceEditionCode }?.topics

    open fun getMetaData(serviceCode: String, serviceEditionCode: String): Map<String, String> =
        routes.find { it.serviceCode == serviceCode && it.serviceEditionCode == serviceEditionCode }?.metadata ?: emptyMap()
}
